From: Sebastian Noack <sebastian.noack@gmail.com>
Date: Wed, 4 Oct 2017 22:38:25 -0400
Subject: Build WLC from subdirectory

Sway is based on WLC, a library that isn't packaged in any Debian release.
In order to keep this package self-contained, build WLC into Sway.
---
 CMakeLists.txt         | 8 ++++++--
 wlc/src/CMakeLists.txt | 5 -----
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99081dc..8648d26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_EXTENSIONS OFF)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
-add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result -Werror)
+add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result)
 
 list(INSERT CMAKE_MODULE_PATH 0
 	${CMAKE_CURRENT_SOURCE_DIR}/CMake
@@ -50,7 +50,6 @@ endif()
 
 find_package(JsonC 0.12.1 REQUIRED)
 find_package(PCRE REQUIRED)
-find_package(WLC REQUIRED)
 find_package(Wayland REQUIRED)
 find_package(XKBCommon REQUIRED)
 find_package(Cairo REQUIRED)
@@ -97,6 +96,11 @@ endif()
 
 include_directories(include)
 
+set(WLC_BUILD_STATIC ON CACHE STRING "" FORCE)
+set(WLC_BUILD_EXAMPLES OFF CACHE STRING "" FORCE)
+set(WLC_BUILD_TESTS OFF CACHE STRING "" FORCE)
+add_subdirectory(wlc)
+
 add_subdirectory(protocols)
 add_subdirectory(common)
 add_subdirectory(wayland)
diff --git a/wlc/src/CMakeLists.txt b/wlc/src/CMakeLists.txt
index e3f3903..a40692d 100644
--- a/wlc/src/CMakeLists.txt
+++ b/wlc/src/CMakeLists.txt
@@ -190,8 +190,3 @@ mark_as_advanced(WLC_DEFINITIONS WLC_INCLUDE_DIRS WLC_LIBRARIES)
 
 # Add pkgconfig
 configure_file(wlc.pc.in wlc.pc @ONLY)
-
-# Install rules
-install(TARGETS wlc DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/wlc" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/wlc.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
